subsys/settings/Kconfig

```diff
@@ -9,6 +9,7 @@ menuconfig SETTINGS
 	default n
 	prompt "Enable settings subsystem with non-volatile storage"
 	depends on FILE_SYSTEM || FCB
+	depends on FLASH_PAGE_LAYOUT
```

Comment:
`depends on FILE_SYSTEM || (FCB && FLASH_PAGE_LAYOUT)` 
+
under `config SETTINGS_FCB` change dependency to `(FCB && FLASH_PAGE_LAYOUT)`

Reply:
Finally i recommend change to:
# Only NFFS is supported as FS.
#The reason ins that FatFs doesn't implements fs_rename() API 
depends on (FILE_SYSTEM && FILE_SYSTEM_NFFS) || (FCB && FLASH_PAGE_LAYOUT)

Reply:
fixed